File: src/my_project/interviews/top_150_questions_round_22/ex_137_triangle.py
from typing import List 

class Solution:
    def minimumTotal(self, triangle: List[List[int]]) -> int:
        dp = triangle[-1][:]
        for row in range(len(triangle) - 2, -1, -1):
            for col in range(len(triangle[row])):
                dp[col] = triangle[row][col] + min(dp[col], dp[col + 1])
        return dp[0]

File: src/my_project/interviews_typescript/top_150_questions_round_1/ex_137_triangle.ts
function minimumTotal(triangle: number[][]): number {
    const dp: number[] = [...triangle[triangle.length - 1]];
    for (let row = triangle.length - 2; row >= 0; row--) {
        for (let col = 0; col < triangle[row].length; col++) {
            dp[col] = triangle[row][col] + Math.min(dp[col], dp[col + 1]);
        }
    }
    return dp[0];
};